### PR TITLE
Preprocessing is way too slow

### PR DESCRIPTION
**Pull request by robertoostenveld**
I tested it: it took 3000 ms to filter a segment of 250 ms. So impossible to run in real time.

I noticed that the onlinefilter implementation that @deep-introspection  wrote is not a one-sample update scheme, but simply a sliding window where the window gets updated every time and fully processed again.

I changed the code so that the whole new data block is processed. With a block size of 63 samples this improves the speed with a factor of 63x. It is still slow, but this is fast enough to make it run in real time.

### CODE REVIEW

File: module/preprocessing/preprocessing.py [hunk 1]
```diff
@@ -115,39 +118,48 @@
 if not(highpassfilter is None) and not(lowpassfilter is None):
     fir_poly = firwin(filterorder, cutoff = [lowpassfilter, highpassfilter], window = 'blackmanharris', pass_zero = False)
 
+def onlinefilter(fil_state, in_data):
+    m = in_data.shape[0]
+    n = len(fir_poly)
+    fil_state = np.concatenate((fil_state, np.atleast_2d(in_data)), axis=0)
```
Comment (robertoostenveld) [inline module/preprocessing/preprocessing.py]: note that I am not transposing the data any more, i.e. it always remains samples*channels

**Comment by robertoostenveld** [comment]
@deep-introspection can you please review my suggested changes and merge them? Alternatively, @stephenwhitmarsh could have a look at it.

**Comment by deep-introspection** [comment]
Sorry for low/high pass filter 🙈👍🏼

Don’t get how you can skip all the window.
Did you tried with a playback file or a sine wave generator?

**Comment by robertoostenveld** [comment]
The issue was (and still is, but now to a lesser extent) that you convolve 511+1 samples for every sample. That is a lot of floating point operations. What I am now doing is to convolve 511+N samples, for every block of N samples. If N=63 (i.e. approx 4 updates per second at 250 Hz) then it convolves 511+63 samples four times per second, which is way more relaxed for the CPU.

I have merged it, just to push it forward. I did test (i.e. it runs), but don't have a good diagnostic method yet. The signal plot module of @stephenwhitmarsh does its own filtering. I guess that I have to record and analyze a longer piece of signal with FieldTrip.

**Pull request merged**